pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -185,6 +187,8 @@ func (s *Server) initDNSCerts(hostname, namespace string) error {
 
 	} else {
 		log.Infof("User specified cert provider: %v", features.PilotCertProvider.Get())
+		dnsKeyFile = model.GetOrDefault(defaultTLSServerKey, dnsKeyFile)
```

Comment:
I don't think we should be modifying global variables like this, it will make a lot of confusion

Reply:
These global variables are used heavily in bootstrap/webhook.go for file watching and if we do not update these then the file watch functionality won't work for custom certificates. 

Here these global file path's files are generated for each type of `PilotCertProvider`. We can add a new type for `PilotCertProvider`, which if set disables IstioCA and uses the provided certs instead. wdyt?

Reply:
To be fair, PilotCertProvider is not a stable API. 

Feel free to propose a proper mesh config setting - or better process to auto-select. 

I'm still not clear what you're trying to do.